tests/test_functional.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+
+from pypst import Content, Document, Functional, Set
+
+
+def test_empty_functional():
+    obj = Functional()
+    assert obj.render() == "#{}"
+
+
+def test_nested_functional():
+    obj = Functional(Functional())
+    assert obj.render() == "#{}"
+
+    obj = Functional(Functional("Foo", context=True))
+    assert obj.render() == "#context {Foo}"
+
+    obj = Functional(Functional("Foo"), context=True)
+    assert obj.render() == "#context {Foo}"
```

Comment:
Could be simplified by using `pytest.mark.parametrize`, but that's mostly nitpicking.